flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/ProcTimeInnerJoin.scala

```diff
@@ -44,12 +48,18 @@ import org.apache.flink.util.Collector
 class ProcTimeInnerJoin(
   private val leftStreamWindowSize: Long,
   private val rightStreamWindowSize: Long,
-  private val element1Type: TypeInformation[CRow],
-  private val element2Type: TypeInformation[CRow],
-  private val filterFunc: RichFilterFunction[Row])
-  extends CoProcessFunction[CRow, CRow, CRow] {
+  private val element1Type: TypeInformation[Row],
+  private val element2Type: TypeInformation[Row],
+  private val genJoinFunc: GeneratedFunction[FlatJoinFunction[Row, Row, Row], Row])
```

Comment:
The other functions pass the `code` and `name` of the class and have `@transient` parameter for the `returnType`. Have a look at `CRowFlatMapRunner` for instance